api/schema/facts.go

```diff
@@ -0,0 +1,35 @@
+package schema
+
+type Fact struct {
+	ID          int      `json:"id,omitempty"`
+	Environment int      `json:"environment,omitempty"`
+	Name        string   `json:"name,omitempty"`
+	Value       string   `json:"value,omitempty"`
+	Source      string   `json:"source,omitempty"`
+	Description string   `json:"description,omitempty"`
+	KeyFact     bool     `json:"keyFact,omitempty"`
```

Comment:
Bools and JSON with omitempty are a bit funny. If keyfact is `false`, it will never display the key:value (https://go.dev/play/p/yqUxOfRkaT_G)

But if it is a pointer bool, then it will display correctly if it has been provided, and omit if it is not defined at all (https://go.dev/play/p/oD5i9Rr6ctD)

This starts to get a bit funny in gql though if `null` is a valid return option for this. What is the default being set in the database if no value for keyfact is provided? is it true, false, or null?

The same will apply to the `AddFactInput` type, if I want to add a keyfact as false, but if the default is `true` or `null`, it won't work because a `false` is "empty" with `omitEmpty`.

If you remove `omitEmpty` though (https://go.dev/play/p/s6GdTIDFIM4) then you will always have true or false, without needing to set the bool as a pointer. This may be acceptable for retrieving facts and adding facts though?
 
If there is ever an `updateFact` mutation, then using a pointer boolean would probably be better because it will only set `keyfact` as true/false if it has been explicitly defined in the data in a patch operation, that way there would be no possibility of changing a keyfact from true to false or false to true unless the user decided it to be so.

Reply:
Given that the default value in the db is set to '0' https://github.com/uselagoon/lagoon/blob/main/services/api-db/legacy-migration-scripts/00-tables.sql#L272C1-L272C59 which default to 'false' in Go's bool type (I think). I'm not sure we will have an updateFact mutation since the work it currently works is that all facts are deleted by source prior to adding/updating facts. 

However if we do not use 'omitempty' then any JSON input will always need to have these fields and for keyFact this probably isn't necessary so a regular bool is fine? Since the database default is 0/false for it then I think we are good

Reply:
I guess it comes down to - do we need to differienate between 'nil' and 'false' with a pointer bool? probably not needed

Reply:
To clarify, I will update to:
```
  KeyFact bool `json:"key_fact"`
```